app/services/hyrax/indexes_thumbnails_decorator.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+# OVERRIDE Hyrax v5.0.0rc2 to make collection thumbnails uploadable
+
+module Hyrax
+  module IndexesThumbnailsDecorator
+    # Returns the value for the thumbnail path to put into the solr document
+    def thumbnail_path
+      if object.class == Collection && UploadedCollectionThumbnailPathService.uploaded_thumbnail?(object)
```

Comment:
What about `object.collection?`. We have `Valkyrie::Collection` coming soon, and class level comparison is always dubious.

I strongly prefer `object.is_a?` over `object.class ==` due to `is_a?` working with modules and such things.

Reply:
That's a good idea, so instead of `object.class == Collection` we do `object.collection?` which, if I'm understanding correctly, would also catch cases of `Valkyrie::Collection`, is that correct?